src/main/java/org/folio/service/orders/flows/update/unopen/UnOpenCompositeOrderManager.java

```diff
@@ -184,44 +201,81 @@ private Future<Void> processInventoryOnlyWithHolding(CompositePoLine compPOL, Re
       .mapEmpty();
   }
 
+  private Future<List<Pair<String, String>>> deleteHoldings(String tenantId, List<JsonObject> holdings, RequestContext requestContext) {
+    if (CollectionUtils.isEmpty(holdings)) {
+      return Future.succeededFuture(List.of());
+    }
+    RequestContext newContext = createContextWithNewTenantId(requestContext, tenantId);
+    List<Future<Pair<String, String>>> deletedHoldingIds = new ArrayList<>(holdings.size());
+    holdings.forEach(holding -> {
+      String holdingId = holding.getString(ID);
+      String permanentLocationId = holding.getString(HOLDING_PERMANENT_LOCATION_ID);
+      if (holdingId != null) {
+        deletedHoldingIds.add(inventoryItemManager.getItemsByHoldingId(holdingId, newContext)
+          .compose(items -> {
+            if (items.isEmpty()) {
+              return inventoryHoldingManager.deleteHoldingById(holdingId, true, newContext)
+                .map(v -> Pair.of(holdingId, permanentLocationId));
+            }
+            return Future.succeededFuture();
+          }));
+      }
+    });
+    return collectResultsOnSuccess(deletedHoldingIds)
+      .map(resultDeletedHoldingVsLocationIds -> resultDeletedHoldingVsLocationIds.stream()
+        .filter(pair -> Objects.nonNull(pair) && Objects.nonNull(pair.getKey())).collect(toList()))
+      .map(resultDeletedHoldingVsLocationIds -> {
+        if (logger.isDebugEnabled()) {
+          String deletedIds = resultDeletedHoldingVsLocationIds.stream().map(Pair::getKey).collect(Collectors.joining(","));
+          logger.debug(String.format("Holdings were removed : %s", deletedIds));
+        }
+        return resultDeletedHoldingVsLocationIds;
+      });
+  }
+
+
   private Future<Void> processInventoryOnlyWithItems(CompositePoLine compPOL, RequestContext requestContext) {
     return deleteExpectedPieces(compPOL, requestContext)
       .compose(deletedPieces ->
-        inventoryItemManager.getItemsByPoLineIdsAndStatus(List.of(compPOL.getId()), ItemStatus.ON_ORDER.value(), requestContext)
-          .compose(onOrderItems -> {
-            if (CollectionUtils.isNotEmpty(onOrderItems)) {
-              List<String> itemIds = onOrderItems.stream().map(item -> item.getString(ID)).collect(toList());
-              return inventoryItemManager.deleteItems(itemIds, false, requestContext);
-            }
-            return Future.succeededFuture();
-          })
-          .onSuccess(v -> logger.info("Items and pieces deleted after Un-Open order with id: {}", compPOL.getId()))
-          .onFailure(e -> logger.error("Items and pieces deletion failed after Un-Open order with id: {}", compPOL.getId(), e))
+        GenericCompositeFuture.all(
+          PoLineCommonUtil.getTenantsFromLocations(compPOL)
```

Comment:
if we are working in non ECS env - Location object will not have tenantId populated at all, it means we will not invoke this logic to process inventory

Reply:
`getTenantsFromLocations` list might contain `null` and `createContextWithNewTenantId` will return same context:
```
    if (tenantId == null) {
      return requestContext;
    }
```
That's why we can't use here isEmpty check. Tenant Id list should not contain empty string, only `null` allowed

Reply:
pushed the change, resolving